src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/EnumConverter.cs

```diff
@@ -524,17 +532,15 @@ private static bool ContainsSpecialChar()
 
             foreach (string name in names)
             {
-                foreach (char cha in name)
+                foreach (char cha in s_specialChars)
```

Comment:
I think the inner for loop could be replaced with string.IndexForAny, which is available in older targets as well:

https://learn.microsoft.com/en-us/dotnet/api/system.string.indexofany?view=netstandard-2.0

Reply:
good suggestion!